shared/libraries/opcuatms/opcuatms_client/src/tms_client.cpp

```diff
@@ -31,10 +33,15 @@ daq::DevicePtr TmsClient::connect()
 {
     OpcUaEndpoint endpoint("TmsClient", opcUaUrl);
     endpoint.registerCustomTypes(UA_TYPES_DI_COUNT, UA_TYPES_DI);
-    endpoint.registerCustomTypes(UA_TYPES_TMSBT_COUNT, UA_TYPES_TMSBT);
-    endpoint.registerCustomTypes(UA_TYPES_TMSBSP_COUNT, UA_TYPES_TMSBSP);
-    endpoint.registerCustomTypes(UA_TYPES_TMSDEVICE_COUNT, UA_TYPES_TMSDEVICE);
-    endpoint.registerCustomTypes(UA_TYPES_TMSESP_COUNT, UA_TYPES_TMSESP);
+    endpoint.registerCustomTypes(UA_TYPES_DAQBT_COUNT, UA_TYPES_DAQBT);
+    endpoint.registerCustomTypes(UA_TYPES_DAQBSP_COUNT, UA_TYPES_DAQBSP);
+    endpoint.registerCustomTypes(UA_TYPES_DAQDEVICE_COUNT, UA_TYPES_DAQDEVICE);
+    endpoint.registerCustomTypes(UA_TYPES_DAQESP_COUNT, UA_TYPES_DAQESP);
+    // TODO: Now, every new added vendor spec needs to be inserted here as well.
+    // It would be better that a method would be called which add all genereted nodesets from CMake.
+    // This also applies then to the openDAQ Types...
+    endpoint.registerCustomTypes(UA_TYPES_DAQHBK_COUNT, UA_TYPES_DAQHBK);
```

Comment:
This can be very easily solved with a configure_file cmake command. As mentioned above, any vendor-specific nodesets should provide their own CMakeLists.txt that would append their name and count variables to two CMake variables.

Reply:
Is the CMake Variable as default set to true?